src/app/apps/rup/mapa-camas/views/plan-indicaciones/motivo-suspension/motivo-suspension.component.ts

```diff
@@ -1,37 +1,66 @@
-import { Component, Output, Input, EventEmitter } from '@angular/core';
+import { Component, Output, Input, EventEmitter, OnInit } from '@angular/core';
+import { Observable } from 'rxjs/internal/Observable';
+import { RUPComponent } from 'src/app/modules/rup/components/core/rup.component';
 
 @Component({
     selector: 'motivo-suspension',
     templateUrl: 'motivo-suspension.component.html'
 })
 
-export class MotivoSuspensionComponent {
+export class MotivoSuspensionComponent extends RUPComponent implements OnInit {
 
-    @Input() motivo;
+    @Input() tipo: 'rechazo' | 'suspension';
+    @Input() key: any;
+    @Input() motivo: any;
     @Output() cancelar = new EventEmitter<boolean>();
     @Output() guardar = new EventEmitter<any>();
-    private nuevoMotivo;
+    rechazo$: Observable<any>;
+    private nuevoMotivo = '';
     public editando = false;
+    public rechazo = { key: null, nombre: null };
+
+    ngOnInit() {
+        if (!this.key) {
+            this.key = '';
+            this.editando = true;
+        } else {
+            this.rechazo = { key: this.key, nombre: this.motivo };
+        }
+        this.nuevoMotivo = this.motivo;
+        this.rechazo$ = this.constantesService.search({ source: 'plan-indicaciones:rechazo' });
+    }
 
     guardarSuspension() {
-        this.motivo = this.nuevoMotivo.slice();
-        this.guardar.emit(this.motivo);
-        this.nuevoMotivo = null;
+        if (this.tipo === 'rechazo') {
+            if (this.rechazo.key === 'otro') {
+                this.rechazo.nombre = this.nuevoMotivo;
+            }
+            this.guardar.emit(this.rechazo);
+        } else {
+            this.guardar.emit(this.motivo);
+        }
         this.editando = false;
     }
 
     cancelarSuspension() {
-        if (!this.editando) {
-            this.cancelar.emit(false);
-        }
+        this.cancelar.emit(false);
         this.editando = false;
         this.nuevoMotivo = null;
     }
 
     editar() {
-        if (this.motivo?.length) {
+        if (this.motivo?.length && this.tipo !== 'rechazo') {
             this.nuevoMotivo = this.motivo;
         }
         this.editando = true;
     }
+
+    motivoRequerido() {
+        return (this.tipo !== 'rechazo' && (!this.motivo || this.editando)) || (this.tipo === 'rechazo' && this.rechazo?.key === 'otro');
+    }
+
+    puedeGuardar() {
+        return (this.tipo !== 'rechazo' && this.nuevoMotivo?.length >= 3) ||
+            (this.tipo === 'rechazo' && (this.rechazo?.key && this.rechazo?.key !== 'otro' || this.rechazo?.key === 'otro' && this.nuevoMotivo?.length >= 3));
```

Comment:
Evitar la necesidad de verificar this.rechazo?.key dos veces simplificandolo a this.rechazo?.key !== 'otro'

Reply:
corregido gracias!